scripts/generate-testcases.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/python3
+
+import argparse
+import os
+import sys
+import re
+from jinja2 import Template
+
+
+def parse_args():
+    _parser = argparse.ArgumentParser(description="Generate tclib testcases from kickstart tests")
+    _parser.add_argument("kickstart_tests_dir", metavar="KSTESTS_DIR",
+                         help="Directory with kickstart tests.")
+    _parser.add_argument("--output-dir", "-o", type=str,
+                         metavar="TEST_CASES_DIR",
+                         help="Existing folder for generated test cases.")
+    _parser.add_argument("--template", "-t", type=str, required=True,
+                         metavar="TEST_CASE_TEMPLATE",
+                         help="Template for a test case")
+    return _parser.parse_args()
+
+
+def get_tags_from_line(line):
+    m = re.match(r'^\s*(TESTTYPE=)(.*)', line)
+    if not m:
+        return None
+    _tt, tags = m.groups()
+    m = re.match(r'.*(TESTTYPE:-)(.*)', tags)
+    if m:
+        _tt, tags = m.groups()
+    tags = [tag.strip('"}') for tag in tags.split()]
+    return tags
```

Comment:
Ouch. It can't be helped right now, but... In long term, is it important to have these `TESTTYPE=${TESTTYPE:-"values"}` cases? This convoluted parsing is just the cherry on top of that.

Reply:
The reason is "inheritance" of .sh files for code deduplication (eg bindtomac-team-pre.sh including team-pre.sh). It is not great solution and it will go away if/when we migrate these metadata (test tags) into tclib testcases.